gatsby/content/blog/2022/02/2022-02-22-synapse-1.53.mdx

```diff
@@ -0,0 +1,37 @@
+---
+date: '2022-02-22'
+title: Synapse 1.53 released
+categories:
+- Releases
+author: Brendan Abolivier
+---
+
+Rejoice everyone, [Synapse 1.53](https://github.com/matrix-org/synapse/releases/tag/v1.53.0) is out! Let's have a look at what's new with this release.
+
+## Stabilisation of registration tokens
+
+Registration tokens is a feature introduced in [Synapse 1.42.0](https://github.com/matrix-org/synapse/releases/tag/v1.42.0), which allows homeserver administrators to force their users to use specific tokens when registering. This is similar to Synapse's [registration shared secret](https://github.com/matrix-org/synapse/blob/6f440fd859c411af8c32478b4353f7550619c3bd/docs/sample_config.yaml#L1291-L1294) support, but with added features, such as the possibility to limit how users can be registered with the same token, or to make a token expire. See [the admin API documentation](https://matrix-org.github.io/synapse/v1.53/usage/administration/admin_api/registration_tokens.html) for more information on how to manage registration tokens on your homeserver.
+
+Registration tokens were initially proposed to the Matrix specification in [MSC3231](https://github.com/matrix-org/matrix-doc/pull/3231) by [Callum Brown](https://github.com/govynnus) during their Google Summer of Code internship last summer. The MSC has since been accepted, and released in the stable Matrix specification as of [Matrix v1.2](https://matrix.org/blog/2022/02/02/matrix-v-1-2-release). As a result, its Synapse implementation has been updated to remove support for unstable identifiers. Administrators of homeservers on which the reverse proxy rules explicitly allow the unstable route for this feature need to update their configuration. Same goes for developers of Matrix clients that support this feature. See the [upgrade notes](https://matrix-org.github.io/synapse/v1.53/upgrade#stablisation-of-msc3231) for more information.
+
+## Time-based cache expiry now enabled by default
+
+To avoid being overly intensive on resources by making too many queries to the database, Synapse maintains several in-memory caches to store data it needs to use frequently. However, this comes with the inconvenience that, if Synapse needs to store too much data, these caches can become fairly big and occupy too much space in the host's memory.
+
+Historically, Synapse has been dealing with this issue by having set sizes for each cache, either hardcoded or set in the configuration, and evicting the oldest items when exceeding this size. But relying just on this solution alone comes with its own issues. For example, adding an entry to the cache that Synapse will only rarely use might cause it to evict other entries that it needs to use more often, thus forcing Synapse to query the database for them again.
```

Comment:
```suggestion
Historically, Synapse has dealt with this issue by having set sizes for each cache, either hardcoded or set in the configuration, and evicting the oldest items when exceeding this size. But relying just on this solution alone comes with its own issues. For example, adding an entry to the cache that Synapse will only rarely use might cause it to evict other entries that it needs to use more often, thus forcing Synapse to query the database for them again.
```

Reply:
we haven't done anything to address the fact that a new cache entry might evict more useful entries, so the last sentence here is wrong.

Reply:
I've removed that sentence entirely since it's not needed and a bit confusing anyway.